### PR TITLE
simplify checkObjectNamespaceLabels

### PR DESCRIPTION
**Pull request by yeedove**
**What type of PR is this?**

**What this PR does / why we need it**:
simplify `checkObjectNamespaceLabels`, and reduce loops.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: internal/provider/kubernetes/predicates.go [hunk 1]
```diff
@@ -76,29 +76,18 @@ func (r *gatewayAPIReconciler) checkObjectNamespaceLabels(nsString string) (bool
 	); err != nil {
 		return false, err
 	}
-	return containAll(ns.Labels, r.namespaceLabels), nil
-}
 
-func containAll(labels map[string]string, labelsToCheck []string) bool {
```
Comment (zirain) [inline internal/provider/kubernetes/predicates.go:82]: it's better to move this under a util package and add ut.

Comment (yeedove) [reply internal/provider/kubernetes/predicates.go:82]: Done

**Review by zirain — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]
lgtm, thanks

**Review by zirain — APPROVED** [review]

**Pull request merged**